src/matlab2tikz.m

```diff
@@ -1778,14 +1780,15 @@ function assertRegularAxes(handle)
     Vertices = get(handle,'Vertices');
     
     % 3D vs 2D
-    if size(Vertices,2) == 3
-        m2t.currentAxesContain3dData = true;
+    is3D = m2t.axesContainers{end}.is3D;
```

Comment:
certainly no problem!
just was wondering and saw that you did not do that below